keyboard-listener/src/main/java/com/rw/keyboardlistener/KeyboardUtils.java

```diff
@@ -41,12 +41,20 @@ public void onGlobalLayout()
         float dp = heightDiff/ mScreenDensity;
         boolean isVisible = dp > MAGIC_NUMBER;
 
-        if (mCallback != null && (prevValue == null || isVisible != prevValue)) {
-            prevValue = isVisible;
+        if (mCallback != null && (sPrevValue == null || isVisible != sPrevValue)) {
+            sPrevValue = isVisible;
             mCallback.onToggleSoftKeyboard(isVisible);
         }
     }
 
+    /**
```

Comment:
I don't think we can do it this way. What if there are multiple listeners? We don't want to share the same static variable that tracks the previous value for all of them, do we? I think the better way of doing this, would be to return a handle to the KeyBoardUtils object when adding the listener and then calling a non-static method on that object to get the current value.